LanguageFeatures/Patterns/type_inference_A18_t04.dart

```diff
@@ -32,8 +30,4 @@ main() {
   Expect.throws(() {
     var <String, int>{"key1": x3} = <String, int>{"key1": pi};
   });
-
-  Map<String, int Function(int)> map = {"key1": foo};
-  final <String, int Function(int)>{"key1": x4} = map;
-  x4.expectStaticType<Exactly<int Function(int)>>();
```

Comment:
Should also be preserved: Coercion works already on the initializing expression in line 36, turning `foo` into `foo<int>`. Again, this is the existing coercion mechanism, no patterns involved.